zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -230,7 +230,8 @@ impl<P: consensus::Parameters> RusqliteMigration for WalletMigration2<P> {
         //
 
         transaction.execute_batch(
-            "CREATE TABLE sent_notes_new (
+            "PRAGMA foreign_keys = ON;
```

Comment:
Move this into the `DROP TABLE accounts` part of the batch (and also move the `OFF` command into there as well, if we can safely respect foreign keys when inserting into `accounts_new`).